deps/ncrypto/ncrypto.cc

```diff
@@ -1016,4 +1025,231 @@ int BIOPointer::Write(BIOPointer* bio, std::string_view message) {
   return BIO_write(bio->get(), message.data(), message.size());
 }
 
+// ============================================================================
+// DHPointer
+
+namespace {
+bool EqualNoCase(const std::string_view a, const std::string_view b) {
```

Comment:
```suggestion
constexpr bool EqualNoCase(const std::string_view a, const std::string_view b) {
```

Reply:
Also a fun article by @lemire https://lemire.me/blog/2024/08/03/converting-ascii-strings-to-lower-case-at-crazy-speeds-with-avx-512/

Reply:
Also you can use strcasecmp on unix

Reply:
Right now I'm going to be more focused on making sure the functionality is correct than optimizing. Let's definitely make sure we take a second pass through looking for perf optimizations as we go tho.